flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamGroupAggregate.scala

```diff
@@ -131,32 +134,42 @@ class DataStreamGroupAggregate(
       s"select: ($aggString)"
     val nonKeyedAggOpName = s"select: ($aggString)"
 
-    val processFunction = AggregateUtil.createGroupAggregateFunction(
-      generator,
-      namedAggregates,
-      inputSchema.relDataType,
-      inputSchema.fieldTypeInfos,
-      groupings,
-      queryConfig,
-      tableEnv.getConfig,
-      DataStreamRetractionRules.isAccRetract(this),
-      DataStreamRetractionRules.isAccRetract(getInput))
-
     val result: DataStream[CRow] =
-    // grouped / keyed aggregation
+      // grouped / keyed aggregation
       if (groupings.nonEmpty) {
+        val keyedProcessFunction = AggregateUtil.createGroupAggregateFunction[Row](
```

Comment:
Maybe we can extract line 140:149 into an inner method so that we can avoid code duplication in this class. The inner method may looks like:
```
    def getKeyedProcessFunction[K]: KeyedProcessFunction[K, CRow, CRow] = {
      AggregateUtil.createGroupAggregateFunction[K](
        generator,
        namedAggregates,
        inputSchema.relDataType,
        inputSchema.fieldTypeInfos,
        groupings,
        queryConfig,
        tableEnv.getConfig,
        DataStreamRetractionRules.isAccRetract(this),
        DataStreamRetractionRules.isAccRetract(getInput))
    }
```
we can use the method in the `if else` clause:
```
   if (groupings.nonEmpty) {
        inputDS
        .keyBy(new CRowKeySelector(groupings, inputSchema.projectedTypeInfo(groupings)))
        .process(getKeyedProcessFunction[Row])
        ...
      }
      // global / non-keyed aggregation
      else {
        inputDS
        .keyBy(new NullByteKeySelector[CRow])
        .process(getKeyedProcessFunction[JByte])
        ....
      }
```

Reply:
I'm not sure if abstracting an internal method that contains only one method call is the best, but the code looks really clean. +1 for your proposal.